plugins/modules/win_dns_delegation.ps1

```diff
@@ -0,0 +1,113 @@
+        state      = @{ type = "str"; choices = "absent", "present"; default = "present" }
+        NameServer = @{ type = "str" }
+        IPAddress  = @{ type = "list"; elements = "str" }
+    }
+    supports_check_mode = $true
+}
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$check_mode = $module.CheckMode
+$name = $module.Params.name
+$zone = $module.Params.zone
+$state = $module.Params.state
+$NameServer = $module.Params.NameServer
+$IPAddress = $module.Params.IPAddress
+
+$parms = @{
+    Name          = $zone
+    ChildZoneName = $name
+}
+Try {
+    # Import DNS Server PS Module
+    Import-Module DNSServer
+} Catch {
+    # Couldn't load the DNS Module
+    $module.FailJson("The DNSServer module failed to load properly: $($_.Exception.Message)", $_)
+}
+
+try {
+    $delegationzone = Get-DnsServerZoneDelegation @parms
+} catch {
+    $module.FailJson("Failed to get dns delegation zone. $($_.Exception.Message)", $_)
+}
+
+if ($state -eq "present") {
+    $parms.NameServer = $NameServer
+    $parms.IPAddress = $IPAddress
+    if ($check_mode) { $parms.WhatIf = $check_mode }
+    #check if exist
+    if ($delegationzone) {
+        if ($delegationzone[0].NameServer.RecordData.NameServer.trim(".") -notmatch $nameserver) {
+            try {
+                Add-DnsServerZoneDelegation @parms -PassThru -Confirm:$false
+                $module.Result.changed = $true
+            } catch {
+                $module.FailJson("Failed to add dns delegation zone $($name): $($_.Exception.Message)", $_)
+            }
+
+        } else {
+            #entry exist need to compare
+            $addressv4 = $delegationzone[0].IPAddress.RecordData.IPV4Address.IPAddressToString
+            $addressv6 = $delegationzone[0].IPAddress.RecordData.IPV6Address.IPAddressToString
+            if ($addressv4) {
+                try {
+                    $diffnsipv4 = Compare-Object -ReferenceObject $addressv4 -DifferenceObject $ipaddress
+                } catch {
+                    $module.FailJson("Failed to compare delegation NameServer zone $($name): $($_.Exception.Message)", $_)
+                }
+            }
+            if ($addressv6) {
+                try {
+                    $diffnsipv6 = Compare-Object -ReferenceObject $addressv6 -DifferenceObject $ipaddress
+                } catch {
+                    $module.FailJson("Failed to compare delegation NameServer zone $($name): $($_.Exception.Message)", $_)
+                }
+            }
+            if (($diffnsipv4.count -gt 0) -or ($diffnsipv6.count -gt 0 )) {
+                try {
+                    #changing ip address
+                    Set-DnsServerZoneDelegation @parms -Confirm:$false
+                    $module.Result.changed = $true
+                } catch {
+                    $module.FailJson("Failed to change dns Server zone $($name): $($_.Exception.Message)", $_)
+                }
+            } else {
+                $module.Result.changed = $false
```

Comment:
Just an FYI, `changed = $false` is the default, you typically only ever need to set this value if you are setting it to `$true`.